pkg/executor/push.go

```diff
@@ -101,6 +101,12 @@ func DoPush(image v1.Image, opts *config.KanikoOptions) error {
 	}
 
 	if opts.TarPath != "" {
+		destRef, err := name.NewTag("temp/tag", name.WeakValidation)
```

Comment:
Sorry for a late review, you shd check if there is no `destination` is provided.